packages/flutter_tools/lib/src/windows/visual_studio.dart

```diff
@@ -350,6 +360,19 @@ class VisualStudio {
     visualStudioDetails ??= _visualStudioDetails(
         validateRequirements: true,
         additionalArguments: <String>[...minimumVersionArguments, _vswherePrereleaseArgument]);
+    // If Visual Studio IDE is not found, try searching for Build Tools.
+    visualStudioDetails ??= _visualStudioDetails(
+        validateRequirements: true,
+        additionalArguments: minimumVersionArguments,
+        requiredWorkload: _requiredBuildToolsWorkload
+        );
+    // If Visual Studio IDE and a stable version Build Tools are not found,
+    // try searching for a pre-release version Build Tools.
+    visualStudioDetails ??= _visualStudioDetails(
+        validateRequirements: true,
+        additionalArguments: <String>[...minimumVersionArguments, _vswherePrereleaseArgument],
+        requiredWorkload: _requiredBuildToolsWorkload
+        );
```

Comment:
The combinatorics are getting a bit much for duplicating code at this point. I think a nested loop with the outer controlling whether or not to check for prerelease and the inner controlling the `requiredWorkload` would make it easier to understand what is the same and what is different in each check.